src/main/java/uk/gov/hmcts/ccd/domain/model/definition/SearchResultDefinition.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.domain.model.definition;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class SearchResultDefinition implements Serializable {
+    private SearchResultField[] fields;
+
+    public SearchResultField[] getFields() {
+        return fields;
+    }
+
+    public void setFields(SearchResultField[] fields) {
+        this.fields = fields;
+    }
+
+    public List<SearchResultField> getFieldsWithPaths() {
+        return Arrays.stream(fields)
+            .filter(f -> StringUtils.isNotBlank(f.getCaseFieldPath()))
+            .collect(Collectors.toList());
+    }
+
+    public Map<String, String> getFieldsUserRoles(SearchResultDefinition searchResultDefinition) {
```

Comment:
why you need to pass SearchResultDefinition? you are inside SearchResultDefinition here. Same for the other APIs

Reply:
good spot- agreed I have removed